src/syngen/ml/interface/interface.py

```diff
@@ -139,7 +139,10 @@ def run(
         data, schema = DataLoader(source).load_data()
         # remove completely empty columns
         data = data.dropna(how="all", axis=1)
-        schema = {column: data_type for column, data_type in schema.items() if column in data.columns}
+        schema["fields"] = {
+            column: data_type for column, data_type in schema.get("fields", {}).items() if column in data.columns
```

Comment:
Hi! Will this work if `schema` is None? Cause None is returned for csv

Reply:
Thank you) I'll double-check it. Seems that you are right. In this case, we will have the exception.

Reply:
Thanks for checking! It fails in current version of main as well, so it would be great to fix it here :)